### PR TITLE
test: some more tests for django

### PR DESCRIPTION
**Pull request by gauravsnj**

### CODE REVIEW

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (olavloite) [inline src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java:276]: We should test that null values are correctly mapped for all possible data types, not only for strings. So we should create an ORM model / class that has all data types and then verify that we can select an instance of that model when:
1. All values are non-null
2. All values are null

See for example this test for `gorm`:
1. `gorm` struct for an entity with all data types: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/golang/pgadapter_gorm_tests/gorm.go#L65
2. A test that verifies that selecting an instance of that entity with all null values works as expected: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/java/com/google/cloud/spanner/pgadapter/golang/GormMockServerTest.java#L333

File: src/test/python/django/transaction_test.py
Comment (olavloite) [inline src/test/python/django/transaction_test.py]: We should not raise an exception in the application itself, but simulate an error in Cloud Spanner. The reason for this is that PostgreSQL has a very strict model when it comes to handling database errors in transactions:
1. A database error in a transaction will invalidate that transaction, and all further statements in that transaction will also fail with the error `current transaction is aborted, commands ignored until end of transaction block`.
2. The only way to get out of that state is to roll back the transaction (or roll back to a savepoint in the transaction, but that is not supported in PGAdapter at the moment).

We should therefore verify that PGAdapter in combination with DJango really shows that behavior.

See this example:
1. `gorm` code: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/golang/pgadapter_gorm_tests/gorm.go#L527
2. Corresponding Java test case: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/java/com/google/cloud/spanner/pgadapter/golang/GormMockServerTest.java#L780

**Review by olavloite — CHANGES_REQUESTED** [review]

File: src/test/python/django/transaction_test.py
Comment (gauravsnj) [reply src/test/python/django/transaction_test.py]: Okay got it. Changed the code.

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (gauravsnj) [reply src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java:276]: done

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (olavloite) [inline src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java:359]: nit: could we also have a verification here that the statement is executed using a single-use read-only transaction? So something like `assertTrue(request.getTransaction().hasSingleUse());`

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (olavloite) [inline src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java]: nit: could we add a verification that the statement is executed in auto-commit mode. That is:
1. The execute request includes a BeginTransaction option with a read/write-option=true.
2. There is a commit request on the server.

The reason for the above is that Django is a very big exception to most ORMs. Most ORMs will not use auto-commit by default, but will use read/write transactions that must be committed explicitly by default. We should verify this behavior in order to ensure that our implementation is correct, and to be able to detect potential regressions in the future if we or the Django maintainers make any changes.

**Review by olavloite — APPROVED** [review]
LGTM (once we have been able to fix the test failure)

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (gauravsnj) [reply src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java]: done

File: src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java
Comment (gauravsnj) [reply src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoBasicTest.java:359]: done

**Pull request merged**
